packages/assets-controllers/src/TokensController.test.ts

```diff
@@ -2160,24 +2199,147 @@ describe('TokensController', () => {
       });
     });
   });
+
+  describe('when selectedAccountId is not set or account not found', () => {
+    describe('detectTokens', () => {
+      it('updates the token states to empty arrays if the selectedAccountId account is undefined', async () => {
+        await withController(async ({ controller, changeNetwork }) => {
+          ContractMock.mockReturnValue(
+            buildMockEthersERC721Contract({ supportsInterface: false }),
+          );
+          changeNetwork({ selectedNetworkClientId: InfuraNetworkType.sepolia });
+
+          expect(controller.state.tokens).toStrictEqual([]);
+          expect(controller.state.ignoredTokens).toStrictEqual([]);
+          expect(controller.state.detectedTokens).toStrictEqual([]);
+        });
+      });
+    });
+
+    describe('addToken', () => {
+      it('handles undefined selected account', async () => {
+        await withController(async ({ controller, getAccountHandler }) => {
+          getAccountHandler.mockReturnValue(undefined);
+          const contractAddresses = Object.keys(contractMaps);
+          const erc721ContractAddresses = contractAddresses.filter(
+            (contractAddress) => contractMaps[contractAddress].erc721 === true,
+          );
+          const address = erc721ContractAddresses[0];
+          const { symbol, decimals } = contractMaps[address];
+
+          await controller.addToken({ address, symbol, decimals });
+
+          expect(controller.state.tokens).toStrictEqual([
+            {
+              address: '0x9C8fF314C9Bc7F6e59A9d9225Fb22946427eDC03',
+              aggregators: [],
+              decimals: 0,
+              image:
+                'https://static.cx.metamask.io/api/v1/tokenIcons/1/0x9c8ff314c9bc7f6e59a9d9225fb22946427edc03.png',
+              isERC721: true,
+              name: undefined,
+              symbol: 'NOUN',
+            },
+          ]);
+        });
+      });
+    });
+
+    describe('addDetectedTokens', () => {
+      it('handles an undefined selected account', async () => {
+        await withController(async ({ controller, getAccountHandler }) => {
+          getAccountHandler.mockReturnValue(undefined);
+          await controller.addDetectedTokens([
+            {
+              address: '0x01',
+              symbol: 'barA',
+              decimals: 2,
+              aggregators: [],
+            },
+          ]);
+          console.log(controller.state.allDetectedTokens[ChainId.mainnet]);
+          expect(controller.state.detectedTokens[0]).toStrictEqual({
+            address: '0x01',
+            decimals: 2,
+            image: undefined,
+            symbol: 'barA',
+            aggregators: [],
+            isERC721: undefined,
+            name: undefined,
```

Comment:
Would be great to define variables for `address`, `symbol`, etc.. and use them in both the `.addDetectedTokens` and the `.toStrictEqual`